app/uk/gov/hmrc/taxhistory/model/api/CompanyBenefit.scala

```diff
@@ -18,13 +18,34 @@ package uk.gov.hmrc.taxhistory.model.api
 
 import java.util.UUID
 
-import play.api.libs.json.Json
+import play.api.libs.json.{Json, _}
 
-case class CompanyBenefit(companyBenefitId:UUID = UUID.randomUUID(),
-                     iabdType: String,
-                     amount: BigDecimal,
-                     source: Option[Int] = None)
+case class CompanyBenefit(companyBenefitId: UUID = UUID.randomUUID(),
+                          iabdType: String,
+                          amount: BigDecimal,
+                          source: Option[Int] = None) {
+  def isForecastBenefit: Boolean = {
+    val P11D_ECS = 19
+    val P11D_Manual = 21
+    val P11D_Assessed = 28
+    val P11D_P9D_Amended = 29
+
+    val isP11D = source.contains(P11D_ECS) || source.contains(P11D_Manual) || source.contains(P11D_Assessed) || source.contains(P11D_P9D_Amended)
```

Comment:
I can see your testing them independently, but try with a few that will fail and stick one that is supposed to pass this.